### PR TITLE
Loading improvements

### LINKED ISSUE #71: [Batch Viewer] Improve Loading indicator
**Description**
Improve the loading of the viewer. Right now it shows the loader on the top left corner:

> Thinking at a minimum make the loader centered in the page. For a more fancy version we could make up a specific form of graph loading indicator but that’s not high priority IMO. It might even be possible to get that “for free” depending on the graph lib used I guess.

image

### PR DESCRIPTION
**Pull request by alongoni**
# Summary

Closes #71
image

1. Open the [Transaction details](https://pr80--gpui.review.gnosisdev.com/tx/0x8bb4f9644bdf19684d66c55133ded17c198e36069a502f903311eb07ed978145/) page and open the Batch Viewer. Something like this should appear as a loader:
https://www.loom.com/share/499aa7cda2f449e0b6acf90ce6460e01

### CODE REVIEW

**Comment by ramirotw** [comment]
Holy cow!

What about adding it to the [User details](https://pr80--gpui.review.gnosisdev.com/rinkeby/address/0xEECA07Acf6d34B3b68d841778fE8bCFA3e817d59/) and also to the [Order details](https://pr80--gpui.review.gnosisdev.com/rinkeby/orders/0x051478b6f1bf660498cb2dd788643f4dab258c61f624b13855c2f0f324238078eeca07acf6d34b3b68d841778fe8bcfa3e817d596213f2e0)?

DancingCowGIF

File: src/components/common/CowLoading.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const EyesOpacity = keyframes`
+  from {
+      opacity: 1;
+    }
+    30% {
+      opacity: 0.3;
+    }
+    to {
+      opacity: 1;
+    }
```
Comment (alfetopito) [inline src/components/common/CowLoading.tsx:32]: Nitpick: align code?

**Review by alfetopito — COMMENTED** [review]
It's loading too fast to me so I cannot see the animation for very long, so I've tried the storybook instead


https://user-images.githubusercontent.com/43217/166647063-75b18caa-6e4f-4e87-8875-df17bb6ca084.mov

Is it because of color theme that I cannot notice the effect in the eyes?
I mean, here the eyes are black and I only see the right one bouncing a little

**Comment by elena-zh** [comment]
Hey @alongoni , great job!

Fust a nitpick: I noticed, that some elements are still loading in tables when the 'loading cow' disappears. See the video: https://watch.screencastify.com/v/apygTfJrmAvlDxeCshbC
Can we 'expand' this loading effect somehow and show it until all page elements are loaded? 

Thanks!

**Review by anxolin — COMMENTED** [review]
I love the initiative! :) 

There's a strange effect when loading. First the loader shows in one position, then it quickly charges, and the it disappears. It creates a strange effect.

I would suggest fixing this logo jumping, and maybe experiment with a delay, so if it loads too quickly we don't even show the loader. Or it appears using a CSS animation so it doesn't do this effects.

**Comment by ramirotw** [comment]
> Fust a nitpick: I noticed, that some elements are still loading in tables when the 'loading cow' disappears. See the video: https://watch.screencastify.com/v/apygTfJrmAvlDxeCshbC image
> 
> Can we 'expand' this loading effect somehow and show it until all page elements are loaded?

This @elena-zh implies making other code related changes that weren't part of this ticket, if the intention is to replace or hide somehow the intermediate spinners we should do it throughout the app but that also requires more thinking and it makes sense to do it in another ticket so @alongoni can focus on the style behavior. Wdyt?

**Comment by elena-zh** [comment]
@ramirotw , I have added this comment to the https://github.com/cowprotocol/explorer/issues/21#issuecomment-1117177876 issue

**Comment by ramirotw** [comment]
@alongoni could you take a look into the [build error](https://github.com/cowprotocol/explorer/runs/6294973782?check_suite_focus=true)?

```
ERROR in /home/runner/work/explorer/explorer/src/apps/explorer/components/TransactionsTableWidget/index.tsx
ERROR in /home/runner/work/explorer/explorer/src/apps/explorer/components/TransactionsTableWidget/index.tsx(9,1):
TS6133: 'Spinner' is declared but its value is never read.
```

**Comment by henrypalacios** [comment]
@alongoni  I have put the `CowLoading` component also in the `tab` **Orders** to be consistent with all other views.

Selection_520

In the future it may be possible to use shimming with the shape of the content as we did with the volume graph and use `Cow loading` only for the first rendering.

**Review by henrypalacios — APPROVED** [review]

**Comment by alfetopito** [comment]
> @alongoni I have put the `CowLoading` component also in the `tab` **Orders** to be consistent with all other views.
> 
> Selection_520
> 
> In the future it may be possible to use shimming with the shape of the content as we did with the volume graph and use `Cow loading` only for the first rendering.

I like this idea, could you create an issue for that so we don't forget to get back to it?

**Pull request merged**